### ISSUE TITLE
8chan SSL/HTTPS Error

### ISSUE THREAD

**Issue by Udinanon** [miku-nyan/Overchan-Android#51]
8chan seems to not be working correctly, as it continues to give a SSL/HTTPS Error without any cause, also the site is perfectly accessible via browser.

**Comment by miku-nyan** [comment]
Overchan doesn't include any CA certificates. So maybe certificates in your system (Android) are outdated and 8chan certificate is not acceptable. In this case you can just turn on "Ignore SSL Errors" option: Preferences -> Imageboards -> 8chan.
(maybe in the future I will try to add support for installing custom trusted certificates)

**Comment by Udinanon** [comment]
Thanks for the solution, but I'm not understanding how does this work
Why did it give this problem all of the sudden? 
Why now and not before? 
Why did it also spread to 4chan on my session? 
Also, can the certificate be outdated and not work in overchan but still be accepted in the Chrome app? 
Sorry and thanks

**Comment by miku-nyan** [comment]
What version of Android are you using?
Are you using https (or plain http) in the Chrome?

**Comment by Udinanon** [comment]
I'm on lollipop 5.1.1  and Chrome is using https

**Comment by patosullivan** [comment]
I'm having the same issue on Android 5.1, just started about a week ago. My fiancee is also having the same issue, running the same version of Android.

Just upgraded to Overchan 1.3 and the issue persists.

**Comment by patosullivan** [comment]
It also started with 8chan and the later occurred with 4chan as well.

**Comment by miku-nyan** [comment]
I think I reproduced this bug.
It happens when last Google Play services installed. (Overchan uses the SSL provider from Play Services package when it is available)

On some sites, for some reason, there is an exception during the certificate validation: `java.security.NoSuchAlgorithmException: No provider found for ecdsa-with-SHA384`
Full stacktrace example:

```
javax.net.ssl.SSLHandshakeException: java.security.cert.CertPathValidatorException: TrustAnchor found but certificate validation failed.
    at com.google.android.gms.org.conscrypt.OpenSSLSocketImpl.startHandshake(SourceFile:332)
    at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:340)
    at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:281)
    at org.apache.http.impl.conn.HttpClientConnectionOperator.connect(HttpClientConnectionOperator.java:124)
    at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:318)
    at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:373)
    at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:225)
    at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:195)
    at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:86)
    at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:108)
    at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:178)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:106)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:57)
    at nya.miku.wishmaster.http.client.HttpClientWrapper.execute(HttpClientWrapper.java:58)
    at nya.miku.wishmaster.http.streamer.HttpStreamer.getFromUrl(HttpStreamer.java:161)
    at nya.miku.wishmaster.http.streamer.HttpStreamer.getJSONFromUrl(HttpStreamer.java:341)
    at nya.miku.wishmaster.http.streamer.HttpStreamer.getJSONObjectFromUrl(HttpStreamer.java:313)
    at nya.miku.wishmaster.chans.fourchan.FourchanModule.downloadJSONObject(FourchanModule.java:137)
    at nya.miku.wishmaster.chans.fourchan.FourchanModule.getThreadsList(FourchanModule.java:421)
    at nya.miku.wishmaster.api.util.PageLoaderFromChan.run(PageLoaderFromChan.java:91)
    at nya.miku.wishmaster.ui.presentation.BoardFragment$PageGetter.loadFromChan(BoardFragment.java:1264)
    at nya.miku.wishmaster.ui.presentation.BoardFragment$PageGetter.run(BoardFragment.java:1129)
    at java.lang.Thread.run(Thread.java:818)
Caused by: java.security.cert.CertificateException: java.security.cert.CertPathValidatorException: TrustAnchor found but certificate validation failed.
    at com.android.org.conscrypt.TrustManagerImpl.checkTrusted(TrustManagerImpl.java:344)
    at com.android.org.conscrypt.TrustManagerImpl.checkServerTrusted(TrustManagerImpl.java:209)
    at com.google.android.gms.org.conscrypt.Platform.checkServerTrusted(SourceFile:166)
    at com.google.android.gms.org.conscrypt.OpenSSLSocketImpl.verifyCertificateChain(SourceFile:560)
    at com.google.android.gms.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
    at com.google.android.gms.org.conscrypt.OpenSSLSocketImpl.startHandshake(SourceFile:328)
    ... 23 more
Caused by: java.security.cert.CertPathValidatorException: TrustAnchor found but certificate validation failed.
    at com.android.org.bouncycastle.jce.provider.PKIXCertPathValidatorSpi.engineValidate(PKIXCertPathValidatorSpi.java:155)
    at java.security.cert.CertPathValidator.validate(CertPathValidator.java:191)
    at com.android.org.conscrypt.TrustManagerImpl.checkTrusted(TrustManagerImpl.java:331)
    ... 28 more
Caused by: com.android.org.bouncycastle.jce.provider.AnnotatedException: TrustAnchor found but certificate validation failed.
    at com.android.org.bouncycastle.jce.provider.CertPathValidatorUtilities.findTrustAnchor(CertPathValidatorUtilities.java:235)
    at com.android.org.bouncycastle.jce.provider.PKIXCertPathValidatorSpi.engineValidate(PKIXCertPathValidatorSpi.java:150)
    ... 30 more
Caused by: java.security.NoSuchAlgorithmException: No provider found for ecdsa-with-SHA384
    at java.security.Signature.getSignature(Signature.java:183)
    at java.security.Signature.getInstance(Signature.java:105)
    at com.google.android.gms.org.conscrypt.OpenSSLX509Certificate.verifyInternal(SourceFile:367)
    at com.google.android.gms.org.conscrypt.OpenSSLX509Certificate.verify(SourceFile:388)
    at com.android.org.bouncycastle.jce.provider.CertPathValidatorUtilities.verifyX509Certificate(CertPathValidatorUtilities.java:1347)
    at com.android.org.bouncycastle.jce.provider.CertPathValidatorUtilities.findTrustAnchor(CertPathValidatorUtilities.java:222)
    ... 31 more
```

Temporary fix (in Overchan 1.3.2): do not use this security provider for Android 5+ (they already support all SSL protocols anyway).
https://github.com/miku-nyan/Overchan-Android/commit/d734eab3e322f254205acf544dc64ffa4b466167

**Comment by patosullivan** [comment]
Yep, this commit resolved the issue.

Thank you!
